ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorTest.java

```diff
@@ -118,4 +120,22 @@ public void shouldRedirectQueriesToQueryEndPoint() {
       );
     }
   }
+
+  @RunWith(MockitoJUnitRunner.class)
+  public static class RateLimit {
+
+    @Rule
+    public final TemporaryEngine engine = new TemporaryEngine()
+        .withConfigs(ImmutableMap.of(KsqlConfig.KSQL_QUERY_PULL_MAX_QPS_CONFIG, 2));
+
+    @Test
+    public void shouldRateLimit() {
```

Comment:
How does this fail here? There are no queries issued?

Reply:
It checks the limit before the request is issued.